dbt/adapters/bigquery/connections.py

```diff
@@ -305,6 +305,7 @@ def get_impersonated_bigquery_credentials(cls, profile_credentials):
         )
 
     @classmethod
+    @retry.Retry()  # google decorator. retries on transient errors with exponential backoff
```

Comment:
_Live discussion_

This is Google's own `Retry()` context manager. It has really sensible defaults and behavior (much better than ours!), and we don't think there's any risk in just putting it everywhere. The scope of the current work is standardizing how we open connections, though, and this is the place to do that.

Follow-up work:
- Add this `Retry()` decorator in more places (specifically, in place of our current homespun approach via `_retry_and_handle`)
- If the user provides configuration for the `job_*` retry/timeout options, plumb them into the `Retry()` decorator in the appropriate ways. It is reasonable that a user would want a very quick timeout if a job fails to create/submit, and a much much _much_ longer timeout for a job that's actually running (= models that take >1 hr to run).

Reply:
this is captured in #231 now

Reply:
reposting from [#231-comment](https://github.com/dbt-labs/dbt-bigquery/issues/231#issuecomment-1231026219)

I agree/like this approach. it's much cleaner and leverage the more robust Google Retry mechanism.

One thing to call out.
Dbt currently has two ways to config bq job retry: `job_retry_deadline_seconds` and `job_retries`. 

However, [Google Retry](https://googleapis.dev/python/google-api-core/latest/retry.html#google.api_core.retry.Retry) only uses `job_retry_deadline_seconds`, to control retries, not `job_retries`. 

when moving over to Google Retry, I suggest 
 - removing `job_retries` config
 - expand `job_retry_deadline_seconds` into two configs: `job_creation_retry_deadline_seconds` and `job_execution_retry_deadline_seconds`
this provide the proper control over retries in both `job_creation` and `job_execution` steps.